src/lib.rs

```diff
@@ -948,6 +946,14 @@ pub fn thread_rng() -> ThreadRng {
     ThreadRng { rng: THREAD_RNG_KEY.with(|t| t.clone()) }
 }
 
+fn weak_seed() -> [usize; 1] {
+    let now = time::SystemTime::now();
+    let unix_time = now.duration_since(time::UNIX_EPOCH).unwrap();
+    let secs = unix_time.as_secs() as usize;
+    let nanos = unix_time.subsec_nanos() as usize;
+    [(secs << 32) | nanos]
```

Comment:
1s = 10^9 ns, not 2^32 ns

It doesn't matter precisely, but if you want a fast approximation use `<< 30` to avoid leaving 2 empty bits